tests/fail/tree_borrows/write_to_shr.rs

```diff
@@ -0,0 +1,11 @@
+//@compile-flags: -Zmiri-tree-borrows
+
+// Very simple test that a shared reference without interior mutability cannot
+// be written to.
+
+pub fn main() {
+    let x = &mut 0u64;
+    let xref = unsafe { &*(x as *mut u64) };
+    let xmut = unsafe { &mut*(xref as *const u64 as *mut u64) };
+    *xmut = 31; //~ ERROR: /write access through .* is forbidden/
+}
```

Comment:
This is a duplicate of shr_frozen_violation1, isn't it?

Reply:
Indeed